### PR TITLE
docs/Versions: Cleanup and clarifications

### PR DESCRIPTION
**Pull request by gromgit**
* Removed all mentions of `homebrew/versions`, since it's dead and long gone.
* Version support requirement: Updated PHP example, and added contrasting example to address a common misconception
* Standardize on the use of "main" rather than "non-versioned" to avoid confusion
* Directly address the most common reason for (disallowed) versioning

*Q:* In "Versioned formulae should be as similar as possible and sensible compared to their parent formulae.", what's the intended meaning of "sensible"?

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
LGTM

File: docs/Versions.md [hunk 1]
```diff
@@ -1,26 +1,24 @@
...
-* Versioned formulae should only be linkable at the same time as their non-versioned counterpart if the upstream project provides support for it, e.g. using suffixed binaries. If this is not possible, use `keg_only :versioned_formula` to allow users to have multiple versions installed at once.
-* A `keg_only :versioned_formula` should not `post_install` anything in the `HOMEBREW_PREFIX` that conflicts with or duplicates the non-versioned counterpart (or other versioned formulae). For example, a `node@6` formula should not install its `npm` into `HOMEBREW_PREFIX` like the `node` formula does.
-* Versioned formulae submitted should be expected to be used by a large number of people. If this ceases to be the case: they will be removed. We will aim not to remove those in the [top 3,000 `install_on_request` formulae](https://brew.sh/analytics/install-on-request/).
+* Versioned formulae should only be linkable at the same time as their main formula if the upstream project provides support for it, e.g. using suffixed binaries. If this is not possible, use `keg_only :versioned_formula` to allow users to have multiple versions installed at once.
```
Comment (MikeMcQuaid) [inline docs/Versions.md:14]: I prefer `non-versioned` to `main`; I think it's less ambiguous.

File: docs/Versions.md [hunk 2]
```diff
@@ -1,26 +1,24 @@
...
 
-Homebrew's versions are not intended to be used for any old versions you personally require for your project. You should create your own [tap](How-to-Create-and-Maintain-a-Tap.md) for formulae you or your organisation wish to control the versioning of or those that do not meet the above standards. Software that has regular API or ABI breaking releases still needs to meet all the above requirements; that a `brew upgrade` has broken something for you is not an argument for us to add and maintain a formula for you.
+**Homebrew's versions should not be used to "pin" formulae to your personal requirements.** You should instead create your own [tap](HowVersioning support-to-Create-and-Maintain-a-Tap.md) for formulae you or your organisation wish to control the versioning of, or those that do not meet the above standards. Software that has regular API or ABI breaking releases still needs to meet all the above requirements; that a `brew upgrade` has broken something for you is not an argument for us to add and maintain a formula for you.
```
Comment (MikeMcQuaid) [inline docs/Versions.md:21]: I don't think this warrants being bold when nothing else on this page is. `HowVersioning` seems broken?

File: docs/Versions.md (hunk 1 above)
Comment (gromgit) [reply docs/Versions.md:14]: OK, I'll revert that one.

File: docs/Versions.md (hunk 2 above)
Comment (gromgit) [reply docs/Versions.md:21]: Noted, and oops.

**Comment by jonchang** [comment]
Squashed commits, will merge when CI is green.

**Comment by jonchang** [comment]
Thanks @gromgit!

**Pull request merged**